File: backend/Shared/EventPlanR.Application/Dto/Common/AddressDto.cs
namespace EventPlanr.Application.Dto.Common;

public class AddressDto
{
    public string Country { get; set; } = null!;
    public string ZipCode { get; set; } = null!;
    public string City { get; set; } = null!;
    public string AddressLine { get; set; } = null!;
}

File: frontend/lib/data/network/models/event_planr/common/address_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'address_dto.freezed.dart';
part 'address_dto.g.dart';

@freezed
class AddressDto with _$AddressDto {
  const factory AddressDto({
    required String country,
    required String zipCode,
    required String city,
    required String addressLine,
  }) = _AddressDto;

  factory AddressDto.fromJson(Map<String, Object?> json) =>
      _$AddressDtoFromJson(json);
}
